catalyst-gateway/bin/src/db/index/queries/staked_ada/get_assets_by_stake_address.rs

```diff
@@ -1,46 +1,54 @@
 //! Get assets by stake address.
-use std::sync::Arc;
+use std::sync::{Arc, LazyLock};
 
-use cardano_blockchain_types::{Slot, StakeAddress};
-use scylla::{
-    prepared_statement::PreparedStatement, transport::iterator::TypedRowStream, DeserializeRow,
-    SerializeRow, Session,
-};
+use cardano_blockchain_types::StakeAddress;
+use futures::TryStreamExt;
+use scylla::{prepared_statement::PreparedStatement, DeserializeRow, SerializeRow, Session};
 use tracing::error;
 
-use crate::db::{
-    index::{
-        queries::{PreparedQueries, PreparedSelectQuery},
-        session::CassandraSession,
+use crate::{
+    db::{
+        index::{
+            queries::{PreparedQueries, PreparedSelectQuery},
+            session::CassandraSession,
+        },
+        types::{DbSlot, DbStakeAddress, DbTxnIndex, DbTxnOutputOffset},
     },
-    types::{DbSlot, DbStakeAddress, DbTxnIndex, DbTxnOutputOffset},
+    settings::Settings,
 };
 
 /// Get assets by stake address query string.
 const GET_ASSETS_BY_STAKE_ADDRESS_QUERY: &str =
     include_str!("../cql/get_assets_by_stake_address.cql");
 
+/// In memory cache of the Cardano native assets data
+static ASSETS_CACHE: LazyLock<
+    moka::future::Cache<DbStakeAddress, Vec<GetAssetsByStakeAddressQuery>>,
+> = LazyLock::new(|| {
+    moka::future::Cache::builder()
```

Comment:
As far as I can see, the default eviction policy is `TinyLfu` and we use `Lru` in other places. Is it deliberate decision?

Reply:
Yea, you are right, we want to use LRU here, will set it explicitly.